### PR TITLE
test: fix changed files detection for external PRs

### PR DESCRIPTION
**Pull request by fiam**
Use a simpler approach to determine the refs that define the interval
of the changes contained in either the PR of the push.
For pull requests, use github.event.pull_request.base.sha and for
pushes just compare HEAD against its previous commit. This should
for the both use cases we support with external repos.

### CODE REVIEW

File: .github/workflows/packages-ci.yaml [hunk 1]
```diff
@@ -65,22 +65,25 @@ jobs:
...
-      - name: Get branch name
-        id: branch-name
-        uses: tj-actions/branch-names@v6
-
-      - name: Get the last successful commit for the current branch
-        uses: nrwl/nx-set-shas@v3
-        id: last_successful_commit_push
-        with:
-          main-branch-name: ${{ steps.branch-name.outputs.current_branch }} # Get the last successful commit for the current branch.
+      - name: Determine base and current refs
```
Comment (StarpTech) [inline .github/workflows/packages-ci.yaml:68]: How did you test it? Did you create an external PR?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: .github/workflows/packages-ci.yaml (hunk 1 above)
Comment (fiam) [reply .github/workflows/packages-ci.yaml:68]: Yep, I tested it in this PR I sent from a personal fork https://github.com/wundergraph/wundergraph/pull/869

**Review by DaniAkash — APPROVED** [review]

**Review by StarpTech — APPROVED** [review]
LGTM

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**